src/groupByKey.ts

```diff
@@ -2,6 +2,15 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((groups: GroupsMap<T>, item) => {
+    const groupKey = item[key] as string;
```

Comment:
The line `const groupKey = item[key] as string;` assumes that the value of `item[key]` is always a string. This may not be the case if the key corresponds to a non-string value. Consider using a more flexible approach, such as converting the value to a string using `String(item[key])` to ensure compatibility with non-string values.